app/src/main/java/com/codesoom/assignment/controllers/TaskController.java

```diff
@@ -0,0 +1,52 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.Task;
+import com.codesoom.assignment.TaskList;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+/**
+ * Handle requests about task.
+ */
+@RestController
+@RequestMapping("/tasks")
+@CrossOrigin
+public class TaskController {
+    TaskList taskList = new TaskList();
+
+    @GetMapping
+    public List<Task> list() {
+        return taskList.getTaskList();
+    }
+
+    @GetMapping("/{id:[0-9]+}")
+    public Task detail(@PathVariable Long id) {
+        return taskList.getTask(id);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public Task create(@RequestBody Task task) {
+        return taskList.add(task);
+    }
+
+    @RequestMapping(path = "/{id:[0-9]+}", method = {RequestMethod.PUT, RequestMethod.PATCH})
+    public Task update(@PathVariable Long id, @RequestBody Task task) {
+        return taskList.modify(id, task.getTitle());
+    }
+
+    @DeleteMapping("/{id:[0-9]+}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public String delete(@PathVariable Long id) {
+        if (taskList.remove(id)) {
+            return "Deleted";
+        }
+
+        throw new ResponseStatusException(
+                HttpStatus.NOT_FOUND, "Fail Deleted"
+        );
+    }
```

Comment:
주어진 `id`에 해당하는 할 일이 없을 경우 `Not Found`를 돌려주고, 있을 경우에는 성공으로 반환하는게 더 자연스러워 보입니다.
```suggestion
    @DeleteMapping("/{id:[0-9]+}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public String delete(@PathVariable Long id) {
        if (!taskList.remove(id)) {
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Fail Deleted");
        }

       return "Deleted";
    }
```

이렇게 할 수 있을까요?

Reply:
네 가능하다고 생각합니다